optaplanner-examples/src/main/java/org/optaplanner/examples/machinereassignment/domain/MrBalancePenalty.java

```diff
@@ -1,18 +1,21 @@
 package org.optaplanner.examples.machinereassignment.domain;
 
-import org.optaplanner.examples.common.domain.AbstractPersistable;
+import org.optaplanner.examples.common.domain.AbstractPersistableJackson;
 
-import com.thoughtworks.xstream.annotations.XStreamAlias;
-
-@XStreamAlias("MrBalancePenalty")
-public class MrBalancePenalty extends AbstractPersistable {
+public class MrBalancePenalty extends AbstractPersistableJackson {
 
     private MrResource originResource;
     private MrResource targetResource;
     private int multiplicand;
     private int weight;
 
-    public MrBalancePenalty() {
+    @SuppressWarnings("unused")
+    MrBalancePenalty() {
+        // required by Jackson
+    }
+
+    public MrBalancePenalty(long id) {
```

Comment:
One of the goals of my ID refactor was to reduce the endless mutability of these classes. These classes are designed to be initialized by setters, which - in my opinion - makes the code too verbose and error-prone.

If at all possible, please remove these newly added constructors and initialize the classes via their full constructors. In some cases, that may not be possible, but in most cases, it will be.